core/src/main/java/org/apache/iceberg/util/PartitionUtil.java

```diff
@@ -51,6 +56,20 @@ private PartitionUtil() {
         MetadataColumns.FILE_PATH.fieldId(),
         convertConstant.apply(Types.StringType.get(), task.file().path()));
 
+    // add _spec_id
+    idToConstant.put(
+        MetadataColumns.SPEC_ID.fieldId(),
+        convertConstant.apply(Types.IntegerType.get(), task.file().specId()));
+
+    // add _partition
+    if (partitionType != null && partitionType.fields().size() > 0) {
+      StructLike coercedPartition = coercePartition(partitionType, spec, partitionData);
+      idToConstant.put(MetadataColumns.PARTITION_COLUMN_ID, convertConstant.apply(partitionType, coercedPartition));
+    } else if (partitionType != null) {
```

Comment:
It seems strange to check `partitionType != null` twice rather than nesting.

Reply:
Fixed.